managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedger.java

```diff
@@ -450,6 +450,23 @@
      */
     Map<String, String> getProperties();
 
+    /**
+     * Add key-value to propertiesMap.
+     *
+     * @param key
+     * @param value
+     * @throws InterruptedException
+     */
+    void setProperty(String key, String value) throws InterruptedException;
```

Comment:
We also need the `asyncSetProperty()` and `asyncDeleteProperty()` versions.

The throws, should also include `ManagedLedgedException`

Reply:
Done.